src/vaadin-form-layout.html

```diff
@@ -296,7 +296,9 @@
 
           this.__mutationObserver = new MutationObserver(mutationRecord => {
             mutationRecord.forEach(mutation => {
-              if (mutation.type === 'attributes' && mutation.attributeName === 'colspan') {
+              if (mutation.type === 'attributes'
+                && ((mutation.attributeName === 'colspan')
+                || (mutation.attributeName === 'hidden' && getComputedStyle(mutation.target).display !== 'none'))) {
```

Comment:
The style update should also take place when an element gets hidden, not just unhidden

Reply:
What I mean is that you can drop the `getComputedStyle(mutation.target).display !== 'none'` because it's an unnecessary check

Reply:
changed